tests/integration/test_workspace_lifecycle.py

```
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+import subprocess
+
+import pytest
+
+from conftest import call_mcp
+
+
+def _list_containers_for_chat(chat_id: str) -> list[dict]:
+    import json
+    r = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)